pkg/stream/client.go

```diff
@@ -0,0 +1,56 @@
+package stream
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const (
+	abortEndpoint = "/abort"
+)
+
+// Client specifies the interface for the
+// client to interact with Stream service.
+type Client interface {
+	AbortChecks(ctx context.Context, checks []string) error
+}
+
+type client struct {
+	URL string
+}
+
+// NewClient builds a new stream client.
+func NewClient(URL string) Client {
+	return &client{URL}
+}
+
+type abortChecksReq struct {
+	Checks []string `json:"checks"`
+}
+
+func (c *client) AbortChecks(ctx context.Context, checks []string) error {
+	reqBody, err := json.Marshal(abortChecksReq{checks})
+	if err != nil {
+		return err
+	}
+	url := fmt.Sprint(c.URL, abortEndpoint)
+	resp, err := http.Post(url, "application/json", bytes.NewBuffer(reqBody))
+	if err != nil {
+		return err
+	}
+	if !isOKStatus(resp.StatusCode) {
```

Comment:
Yes, because I was doubting if stream should return a 201 or a 200 for a successful abort request, but yes, with current implementation we could set it to match 200 only. We can change it if you want :+1:

Reply:
It is okey I just wanted to be sure I wasn't losing something here, no need to change.